### DISCUSSION TITLE
Inquiry about Deepgram STT functionality

### DISCUSSION THREAD

**Discussion by samueljazzjohn** [deepgram/community#252] [General help] [answered]
Does the Deepgram Streaming API provide audio chunking functionality?
Does the Deepgram Streaming API offer built-in transcription access for Google Meet, Zoom, and Teams meetings?
Does the Deepgram Streaming API support separate text retrieval for multiple speakers?

Does the Deepgram Recorded API provide audio chunking functionality?
Does the Deepgram Redorded API offer built-in transcription access for Google Meet, Zoom, and Teams meetings?
Does the Deepgram Recorded API support separate text retrieval for multiple speakers?

**Comment by jkroll-deepgram** [comment]
Hi @samueljazzjohn,

> Does the Deepgram Recorded API provide audio chunking functionality?

You may need to clarify your use case here. Deepgram has a streaming API with an [endpointing](https://developers.deepgram.com/docs/endpointing) feature that chunks audio to detect pauses, in order to return segments of the transcript streamed back to the user. Our pre-recorded API does not chunk audio in a similar manner - we process each audio file in full and return its transcript as a whole.

> Does the Deepgram Recorded API offer built-in transcription access for Google Meet, Zoom, and Teams meetings?

Deepgram does not have built-in integrations with meeting software. Check out our guide [Integrate Deepgram with Zoom](https://developers.deepgram.com/docs/integrate-deepgram-with-zoom) to get a sense of how you can implement this.

> Does the Deepgram Recorded API support separate text retrieval for multiple speakers?

Deepgram offers [Diarization](https://developers.deepgram.com/docs/diarization), which recognizes speaker changes and assigns a speaker to each word in the transcript, such as:

```
[Speaker:0] My name is Beth, and I will be assisting you today. How are you doing?
[Speaker:1] Not too bad. How are you today?
```

**Comment by samueljazzjohn** [comment]
I am creating a bot application that joins a video call at the start of the meeting, records the audio from the ongoing call using one thread, and converts the audio to text using another thread. I would like to know if I can achieve this using Deepgram. Are there any blockers or limitations I should be aware of?


I have successfully recorded and chunked the live audio into binary data, and I have used the Deepgram pre-recorded Python SDK to transcribe it using the deepgram.transcription.prerecorded method with the parameters {"model": "nova", "diarize": "true"}. However, I have noticed that there is a delay of 1.5 to 3 seconds for the chunking of data for each audio file, which is typically between 2 to 10 seconds in length.

**Comment by jkroll-deepgram** [comment, marked as answer]
It sounds like you're utilizing the audio in real time - If so, streaming (`deepgram.transcription.live()` in the Python SDK, [Live Streaming Audio Transcription](https://developers.deepgram.com/docs/python-sdk-streaming-transcription)) will provide lower latency and removes the burden for you to chunk the audio yourself. Streaming also offers [endpointing](https://developers.deepgram.com/docs/endpointing), where you can customize the length of silence for chunking the audio and receiving a response back. You may also find the [Measuring Streaming Latency](https://developers.deepgram.com/docs/measuring-streaming-latency) guide informative.

**Comment by samueljazzjohn** [comment]
```for transcript in transcription_data:
                    transcription_text = transcript['channel']['alternatives'][0]['transcript']
```
Can we use this methode to convert the Json reponse in text. Is there any other way to do that

**Reply by jkroll-deepgram** [reply]
Yes, that is the best way to get the transcript text from the response. If you are using `utterances` or `paragraphs`, you can similarly retrieve sections of the transcript at a per-utterance and per-paragraph level. All of the ways of getting the transcript text will be through parsing the response JSON.

**Comment by samueljazzjohn** [comment]
How to get data directly from the microphone

**Reply by jkroll-deepgram** [reply]
Here's a couple of resources for getting microphone audio in Python:
[How to Transcribe Only What You Need with Python: Listening Before Connected](https://deepgram.com/learn/how-to-transcribe-only-what-you-need-with-python-listening-before-connected)
[Deepgram Streaming Test Suite: Stream Your Microphone](https://github.com/deepgram/streaming-test-suite/tree/main#stream-your-microphone)
